enhancements/microshift/detached-ovn-gateway-bridge.md

```diff
@@ -0,0 +1,486 @@
+- "@trozet, OpenShift Networking"
+- "@zshi-redhat, MicroShift"
+approvers:
+- "@dhellmann"
+api-approvers:
+- None
+creation-date: 2023-04-26
+last-updated: 2023-05-22
+tracking-link:
+- https://issues.redhat.com/browse/NP-654
+see-also:
+- None
+replaces:
+- None
+superseded-by:
+- None
+---
+
+# Use Detached External Gateway Bridge of OVN-Kubernetes for MicroShift
+
+## Summary
+
+This enhancement proposes a new external gateway configuration of OVN-Kubernetes
+for MicroShift. With the new configuration, the `br-ex` will be created without
+binding any host physical interface.
+
+## Motivation
+
+MicroShift is designed to run on IoT and edge devices, it can be deployed on
+various kinds of devices with different network environment. 
+
+Today configure-ovs.sh (microshift-ovs-init.service) sets up OVN-Kubernetes
+gateway bridge `br-ex` for OVN-Kubernetes.  There have been several types of
+issues associated with it mainly due to:
+
+1. transient network interruption when it moves the node IP from uplink port to
+   gateway bridge
+
+2. certain type of network devices are not supported (e.g. wifi, tap/tun)
+
+3. it relies on (default) route info
+
+### User Stories
+
+* As an MicroShift cluster administrator, I want the installation experience of
+MicroShift is similar as a normal application. I don't want to have any
+networking interrupt during the installation.
+
+* As an MicroShift cluster administrator, I want to use a specific network
+  devices (such as wifi, tap/tun) as the node default interface.
+
+* As an MicroShift cluster administrator, I want to install MicroShift on an
+  isolated host that has no default gateway available.
+
+### Goals
+
+* Decouple the br-ex bridge from the host physical interfaces.
+* Enhance OVN-Kubernetes local gateway mode to support external gateway bridge
+  without physical NIC.
+
+### Non-Goals
+
+* Creating a new gateway mode in OVN-Kubernetes for single node cluster.
+
+## Proposal
+
+Instead of creating the br-ex bridge with the node's default interface, we
+create the br-ex without adding any host interface as its uplink port. Today, in
+order to work around the certificates issue, MicroShift adds a static kubernetes
+API server `advertise-address` (default 10.44.0.0/32) to the host `lo`
+interface. To avoid using an extra IP for the dummy gateway interface `br-ex`,
+we will use this advertiseAddress: 
+
+1. if `br-ex` doesn't exist, which means configure-ovs.sh was not invoked, the
+   CNI is not ovn-k, add advertiseAddress to `lo` interface.
+2. Otherwise, add advertiseAddress to `br-ex` interface.
+
+Currently, the local gateway mode and shared gateway mode both use the default
+gateway of the host as the default gateway of the node gateway router. It is
+required by certain egress features like egress IP. However MicroShift doesn't
+need those features . The pod egress traffic will leave OVN-Kubernetes and enter
+host kernel via `mp0` instead of the gateway route. In the context of
```

Comment:
Does `mp0` name have any meaning (convention like `br-ex`)? Or is it just "microshift port"? Like exit port from "Pod network CIDR" to the host?

Reply:
This is a host interface created by OVN-K. The full name is `ovn-k8s-mp0`. It is used for host-to-pod traffic.